packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -694,7 +792,62 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
     return linkMarks;
   }
 
+  private findTextBlocksInRange(
+    node: ProsemirrorNode,
+    range: {
+      from: number;
+      to: number;
+    },
+  ): Array<{ text: string; positionStart: number }> {
+    const nodesWithPos: Array<{
+      node: ProsemirrorNode;
+      pos: number;
+    }> = [];
+    const predicate: (node: ProsemirrorNode) => boolean = (node) =>
+      !node.isTextblock || !node.type.allowsMarkType(this.type);
+
+    // define a placeholder for leaf nodes to calculate link position
+    node.nodesBetween(range.from, range.to, (node, pos) => {
+      if (!predicate(node)) {
+        nodesWithPos.push({
+          node,
+          pos,
+        });
+      }
+    });
```

Comment:
We have a [`findChildren` util](https://github.com/remirror/remirror/blob/ac2019fc59b8caf1f1f8eaaa3163b5a15a459c5c/packages/remirror__core-utils/src/prosemirror-node-utils.ts#L52-L71) you could use here

Reply:
This comes from here https://github.com/remirror/remirror/pull/1817/files#diff-c6362a6560783faf0b832a4b10098233042a3092f87b84b29f28f46c94410ac9L600-L603

```tsx
tr.doc.nodesBetween(from, to, (node, pos) => {
  if (!node.isTextblock || !node.type.allowsMarkType(this.type)) {
    return;
  }

```